packages/flutter_tools/lib/src/web/web_device.dart

```diff
@@ -118,7 +118,7 @@ abstract class ChromiumDevice extends Device {
 
   @override
   Future<LaunchResult> startApp(
-    covariant WebApplicationPackage package, {
+    WebApplicationPackage? package, {
```

Comment:
not directly relevant to the fix, but this type doesn't need to be covariant, as sub-classes will use the exact same type

Reply:
Is this true? Or is it because it has been labeled `covariant` in the base class definition here: https://github.com/flutter/flutter/blob/b70ff4f280ebe0819a9af79e8412a40eb643f5ae/packages/flutter_tools/lib/src/device.dart#L591-L604

Reply:
Is what true? That the type doesn't need to be covariant? Or that it's not relevant to the fix?

Reply:
Is it true that a subtyped class can be passed as a valid parameter to a method.

Meaning I don't think because `WebApplicationPackage` is a subtype of `ApplicationPackage` that we can pass it as a valid type to the `startApp` method. But it works because we marked the type on super class a `covariant`

Reply:
We sync'd offline. To clarify my original statement, I can remove the `covariant` from the `ChromiumDevice.startApp` implementation because the super class has it marked `covariant`. TIL that covariant can be placed in either the superclass or subclass: https://dart.dev/guides/language/sound-problems#the-covariant-keyword